sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -590,6 +591,22 @@ object TypeCoercion {
   }
 
   /**
+   * Coerces NullTypes of a Stack function to the corresponding column types.
+   */
+  object StackCoercion extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveExpressions {
+      case s @ Stack(children @ Literal(_, IntegerType) :: _) if s.childrenResolved =>
```

Comment:
If we check a data type only, `foldable` expression is matched here. So, here, we match `Literal` explicitly.